--- ./extension.cpp	2012-07-12 09:52:25.000000000 -0700
+++ ./extension.new.cpp	2012-07-12 09:47:33.000000000 -0700
@@ -50,6 +50,9 @@
 }
 
 void Extension::LoadModules(Hdf hdf) {
+  if (s_registered_extensions == NULL) {
+    s_registered_extensions = new ExtensionMap();
+  }
   ASSERT(s_registered_extensions);
   for (ExtensionMap::const_iterator iter = s_registered_extensions->begin();
        iter != s_registered_extensions->end(); ++iter) {
